### PR TITLE
RF01TC041_TC_DanielFernandez-04

### PR DESCRIPTION
**Pull request by danielrpg**
UI >Registro de Usuario> Verificar la busqueda de usuarios

### CODE REVIEW

**Comment by danielrpg** [comment]
image

**Comment by danielrpg** [comment]
image

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Usuario/ListUser.java [hunk 1]
```diff
@@ -32,7 +39,20 @@ public FormUser isFormUserVisible() {
         return new FormUser();
     }
 
+
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Usuario/ListUser.java]: remove this empty line

File: src/test/resources/features/users/searchUser.feature [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+  Background: Pagina de inicio es mostrada una vez que se ponen los credenciales correctos de inicio de sesión
+    Given 'SSI-D' pagina de inicio de sesion es cargada
+    And Ingresar los credenciales validos en la pagina 'Login' para ingresar al sistema
+
+  Scenario: Verificar lista de Usuarios este correctamente cargado
+    Given 'Menu principal' estee cargado
+    And seleccionar menu 'Personal' en la pagina 'Menu Principal'
+    And seleccionar submenu 'Usuario' en menu 'Personal'
+    And completar campo 'Buscar Usuario' con los siguientes datos
```
Comment (marbusjim) [inline src/test/resources/features/users/searchUser.feature]: please change to "insertar los siguientes datos en el cuadro de texto 'Buscar Usuario' en la pagina 'Lista de Usuarios'

**Review by jdpapiero — CHANGES_REQUESTED** [review]
review observations

**Review by ivanMise — APPROVED** [review]

File: src/test/resources/features/assignequipament/assignEquipament.feature [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #Show Form for assign equipament
-Feature: Verificar si formulario de asignacion se muestra
+Feature: Verificar formulario de asignacion Equipamiento a Personal
```
Comment (jdpapiero) [inline src/test/resources/features/assignequipament/assignEquipament.feature]: please first word in lowercase

File: src/test/resources/features/users/searchUser.feature [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#Search user
+Feature: check the users search
+
+  Background: Pagina de inicio es mostrada una vez que se ponen los credenciales correctos de inicio de sesión
+    Given 'SSI-D' pagina de inicio de sesion es cargada
+    And Ingresar los credenciales validos en la pagina 'Login' para ingresar al sistema
+
+  Scenario: Verificar lista de Usuarios este correctamente cargado
```
Comment (jdpapiero) [inline src/test/resources/features/users/searchUser.feature]: this title of scenario is not the same to Test Case

File: src/test/resources/features/users/searchUser.feature [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    Given 'Menu principal' estee cargado
+    And seleccionar menu 'Personal' en la pagina 'Menu Principal'
+    And seleccionar submenu 'Usuario' en menu 'Personal'
+    And insertar los siguientes datos en el cuadro de texto 'Buscar Usuario' en la pagina 'Lista de Usuarios'
+      | username |
+      | admin    |
+      | tom      |
+    Then verificar que el resultado coincida con el dato buscado
```
Comment (jdpapiero) [inline src/test/resources/features/users/searchUser.feature:16]: what is the previous step for find user? just set text in input? maybe is necesary to add a step ;)

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 6]
```diff
@@ -305,4 +306,14 @@ public void seleccionarContratos() throws Throwable {
     public void validarListaDeContratos() throws Throwable {
         Assert.assertTrue(listContract.validateContractList(), String.format(ErrorMessage.ERROR_MESSAGE_ELEMENT_VISIBLE, "Contract List"));
     }
+
+    @And("^insertar los siguientes datos en el cuadro de texto 'Buscar Usuario' en la pagina 'Lista de Usuarios'$")
+    public void completarCampoBuscarUsuarioConLosSiguientesDatos(DataTable searchUserTable) throws Throwable {
+        listUser.fillFieldSearchWithData(searchUserTable);
```
Comment (jdpapiero) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: what is the next step after fill data in search field? press enter key or click search?, please add a step to clarify

**Review by jdpapiero — CHANGES_REQUESTED** [review]
plase review observations

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java (hunk 6 above)
Comment (danielrpg) [reply src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: On list users to filter by name is not necessary to click on any button, so is only see data on table list user filtered

File: src/test/resources/features/users/searchUser.feature (hunk 5 above)
Comment (danielrpg) [reply src/test/resources/features/users/searchUser.feature:16]: When you filter user by name you dont need other step for validate information only table of user list

Comment (marbusjim) [reply src/test/resources/features/users/searchUser.feature:16]: If I understand it, when you type a word in the TextField, the table is filtered at the same time, is that true?

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java (hunk 6 above)
Comment (marbusjim) [reply src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: If I understand it, when you type a word in the TextField, the table is filtered at the same time, is that true?

File: src/test/resources/features/users/searchUser.feature (hunk 5 above)
Comment (danielrpg) [reply src/test/resources/features/users/searchUser.feature:16]: Yes is correct for this reason is not necessary  other estep for click other button

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java (hunk 6 above)
Comment (danielrpg) [reply src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: It is related to search field and yes is correct for this reason is not necessary other estep for click other search button

Comment (marbusjim) [reply src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: ok

File: src/test/resources/features/users/searchUser.feature [hunk 7]
```diff
@@ -0,0 +1,16 @@
+#Search user
+Feature: check the users search
```
Comment (marbusjim) [inline src/test/resources/features/users/searchUser.feature]: please use spanish

File: src/test/resources/features/users/searchUser.feature (hunk 5 above)
Comment (marbusjim) [reply src/test/resources/features/users/searchUser.feature:16]: ok

**Review by marbusjim — APPROVED** [review]

**Review by jdpapiero — APPROVED** [review]
Approved

**Pull request merged**